core/src/main/java/hudson/model/Cause.java

```diff
@@ -453,9 +453,12 @@ public String getShortDescription() {
 
         @Override
         public void print(TaskListener listener) {
-            listener.getLogger().println(Messages.Cause_UserIdCause_ShortDescription(
-                    // TODO JENKINS-48467 - better to use ModelHyperlinkNote.encodeTo(User), or User.getUrl, since it handles URL escaping
-                    ModelHyperlinkNote.encodeTo("/user/"+getUserIdOrUnknown(), getUserName())));
+            User user = getUserId() == null ? null : User.getById(getUserId(), false);
+            if(user != null)
+            {
+                listener.getLogger().println(Messages.Cause_UserIdCause_ShortDescription(
+                        ModelHyperlinkNote.encodeTo(user)));
+            }
```

Comment:
If `user == null` it might be appropriate to print some message without a hyperlink, like “started by unknown or anonymous user”. I am not sure if it is important, though—IIRC this message is just used at the beginning of the build log, where it would not be conveying anything of great interest other than that the build was triggered by a user gesture (rather than, say, a schedule or an upstream build). You can always see this detail in the build index page.

Reply:
That makes a lot of sense for log consistency, Ill go ahead and add it / update the tests.